### PR TITLE
[TIMOB-23479] Fix: WebView does not load content in ms-appdata

### PR DESCRIPTION
**Pull request by infosia**
[TIMOB-23479](https://jira.appcelerator.org/browse/TIMOB-23479)

```javascript
var window = Titanium.UI.createWindow(),
    view = Ti.UI.createView({
        backgroundColor: 'blue',
        height: Ti.UI.FILL, width: Ti.UI.FILL
    }),
    webview = Titanium.UI.createWebView({
        height: '80%', width:Ti.UI.FILL, bottom: 0
    }),
    button = Ti.UI.createButton({
        title: 'fromTitanium',
        height: '10%',
        width: Ti.UI.FILL,
        top: 0, left: 0
    });

button.addEventListener('click', function (e) {
    Ti.App.fireEvent('app:fromTitanium', { message: 'event fired from Titanium, handled in WebView' });
});
Ti.App.addEventListener('app:fromWebView', function (e) {
    alert(e.message);
});

// Copy contents to application data directory (ms-appdata://local)
function prepare(files) {
    var webroot = Ti.Filesystem.applicationDataDirectory + 'webroot';
    var webroot_file = Ti.Filesystem.getFile(webroot);
    if (!webroot_file.exists()) {
        webroot_file.createDirectory();
    }
    for (var i = 0; i < files.length; i++) {
        var file = files[i];
        var from = Ti.Filesystem.getFile(Ti.Filesystem.resourcesDirectory, file);
        var to = webroot + Ti.Filesystem.separator + file;
        from.copy(to)
    }
}

window.addEventListener('open', function () {
    prepare(['WebViewIndexPage.htm', 'WebViewNextPage.htm', 'sample.jpg']);
    // load content from ms-appdata
    webview.url = 'ms-appdata:///local/webroot/WebViewIndexPage.htm';

    // load content from application resource
    //webview.url = 'ms-appx-web:///WebViewIndexPage.htm';

    // this loads content from application resource
    //webview.url = 'WebViewIndexPage.htm';
});

view.add(button);
view.add(webview);
window.add(view);
window.open();
```

### Platform-specific Note:

* In order to place content under application data directory, all contents should be placed under sub folder of `local`. (`ms-appdata:///local/your_subfolder/your_content`)
* All contents under data directory (`ms-appdata:///local/your_subfolder`) can be loaded only when index file is placed under data directory (`ms-appdata:///local/your_subfolder/your_index_file.html`).
* In order to hyperlink to the content under `ms-appdata://`, you should use relative path.

```html
    <h2><a href="WebViewNextPage.htm">Follow this link and see if script works</a></h2>
    <img src="sample.jpg" />
```

### CODE REVIEW

File: Source/UI/src/WebView.cpp [hunk 1]
```diff
@@ -17,12 +17,88 @@
...
+					concurrency::event evt;
+					concurrency::create_task(file->OpenAsync(FileAccessMode::Read)).then([&evt, &stream](IRandomAccessStream^ stream_) {
+						stream = stream_;
+						evt.set();
+					}, concurrency::task_continuation_context::use_arbitrary());
+					evt.wait();
+
+					return static_cast<IInputStream^>(stream);
+				});
+				return getInputStreamTask.get();
```
Comment (garymathews) [inline Source/UI/src/WebView.cpp]: Handle any exceptions if a file cannot be loaded

```C++
try {
	return getInputStreamTask.get();
} catch (::Platform::COMException^ e) {
	HAL::detail::ThrowRuntimeError("TitaniumWindows::UI::WebView", "Ti.UI.WebView: Unable to read " + TitaniumWindows::Utility::ConvertString(uri->Path));
}
```

**Comment by garymathews** [comment]
FR **PASS**
CR: **PASS** after exception handling

:+1:

File: Source/UI/src/WebView.cpp (hunk 1 above)
Comment (infosia) [inline Source/UI/src/WebView.cpp]: Nice catch! Updated.

**Comment by infosia** [comment]
build win 10

**Pull request merged**